subprojects/internal-testing/src/main/groovy/org/gradle/test/fixtures/file/TestFile.java

```diff
@@ -274,10 +274,16 @@ public void copyFrom(File target) {
     }
 
     public void copyFrom(URL resource) {
```

Comment:
I extracted the retry logic and use this two retry logic occurrences ([`WatchServiceRegistrar.watchDir`](https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/internal/filewatch/jdk7/WatchServiceRegistrar.java#L155-L191) and [`SocketConnection.writeWithNonBlockingRetry`](https://github.com/gradle/gradle/blob/master/subprojects/messaging/src/main/java/org/gradle/internal/remote/internal/inet/SocketConnection.java#L256-L267)) as extended use case. Would you want me to adapt the extended use case to use `RetryUtil`?